idaes/models/properties/modular_properties/eos/ceos.py

```diff
@@ -447,6 +462,144 @@ def rule_delta_eq(m, p1, p2, p3, i):
+                            (Cubic.gas_constant(b) *
+                            cobj.temperature_crit)**2/cobj.pressure_crit) *
+                        ((1+fw[j]*(1-sqrt(m.temperature_crit_mix /
+                                          cobj.temperature_crit)))**2))
+            b.add_component('a_crit',
+                            Expression(b.component_list,
+                                        rule=func_a_crit,
+                                        doc='Component a coefficient'))
+            
+            def rule_am_crit(m):
+                try:
+                    rule = m.params.get_phase(reference_phase). \
+                        config.equation_of_state_options["mixing_rule_a"]
+                except KeyError:
+                    rule = MixingRuleA.default
+        
+                a_crit = getattr(m, "a_crit")
+                if rule == MixingRuleA.default:
+                    return rule_am_crit_default(m, cname, a_crit)
+                else:
+                    raise ConfigurationError(
+                        "{} Unrecognized option for Equation of State "
+                        "mixing_rule_a: {}. Must be an instance of MixingRuleA "
+                        "Enum.".format(m.name, rule))
+            b.add_component('am_crit', Expression(rule=rule_am_crit))
+        
+            def rule_bm_crit(m):
+                try:
+                    rule = m.params.get_phase(reference_phase). \
+                        config.equation_of_state_options["mixing_rule_b"]
+                except KeyError:
+                    rule = MixingRuleB.default
+        
+                b = getattr(m, cname+"_b")
+                if rule == MixingRuleB.default:
+                    return sum(m.mole_frac_comp[i] * b[i]
+                                for i in m.component_list)
+                else:
+                    raise ConfigurationError(
+                        "{} Unrecognized option for Equation of State "
+                        "mixing_rule_a: {}. Must be an instance of MixingRuleB "
+                        "Enum.".format(m.name, rule))    
+            b.add_component('bm_crit', Expression(rule=rule_bm_crit))
+    
+            def rule_A_crit(m):
+                am_crit = getattr(m, "am_crit")
+                return EoS_param[ctype]['omegaA'] == \
+                    (am_crit * m.pressure_crit_mix /
+                      (Cubic.gas_constant(b) * m.temperature_crit_mix)**2)
+            b.add_component('A_crit', Constraint(rule=rule_A_crit))
+        
+            def rule_B_crit(m):
+                bm_crit = getattr(m, "bm_crit")
+                return EoS_param[ctype]['coeff_b'] == \
+                    (bm_crit * m.pressure_crit_mix /
+                      (Cubic.gas_constant(b) * m.temperature_crit_mix))
+            b.add_component('B_crit', Constraint(rule=rule_B_crit))
+
+    @staticmethod
+    def initialize_critical_properties(b, p):
+        if p == 'Vap':
+            return None
```

Comment:
Hardcoded phase name. Use `b.params.get_phase(p).is_vapor_phase()` instead.

Reply:
Should these be removed from `smooth_VLE.py` as well? For the above instance in `ceos.py` and two instances in `generic_property.py` I used `reference_phase = next(p for p in blk[k].params.phase_list if blk[k].params.get_phase(p).is_liquid_phase())`. Is there a better way to get liquid and vapor phases without explicitly writing 'Liq' etc.?

Reply:
We do need to be a little bit careful here, as there could be multiple vapor and liquid phases. `is_vapor_phase()` will tell you if a phase is vapor or not, but it won't tell you which one it is or help you find others.

However, for now I think `is_vapor_phase()` is the best solution. The possibility of multiple phases of a given type extends far beyond here and needs a lot more though,

Reply:
Okay, I have removed the hardcoded phase names now